contracts/src/compute_engines/yield_curve/YieldCurve.cairo

```diff
@@ -799,7 +799,7 @@ namespace YieldCurve:
 
         # Check that future key is active
         let future_key = future_keys[future_keys_idx]
-        let (future_key_status) = get_future_key_status(spot_entry.key, future_key)
+        let (future_key_status) = get_future_key_status(spot_entry.pair_id, future_key)
```

Comment:
think there are changes needed elsewhere in this file

Reply:
I think it's fine since key and pair_id are logically the same